Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp

```diff
@@ -47,6 +49,9 @@ namespace ROS2::UrdfParser
         }
 
         std::string xmlStr((std::istreambuf_iterator<char>(istream)), std::istreambuf_iterator<char>());
-        return Parse(xmlStr, parserConfig);
+        // modify in memory
+
+        std::string xmlStrChanged = (ROS2::Utils::ModifyURDFInMemory(xmlStr));
```

Comment:
We can make the `Parse*` functions accept an `const SdfAssetBuilderSettings&` instance which can be used to select whether the URDF should be modified in memory before importing.